examples/flow.cpp

```diff
@@ -184,15 +252,17 @@ try
         ioConfig->overrideRestartWriteInterval((size_t)output_interval);
     }
 
-    Opm::BlackoilOutputWriter outputWriter(cGrid,
-                                           param,
-                                           eclipseState,
-                                           pu );
+    const PhaseUsage pu = Opm::phaseUsageFromDeck(deck);
+    Opm::BlackoilOutputWriter outputWriter(grid, param, eclipseState, pu );
 
     // Rock and fluid init
-    props.reset(new BlackoilPropertiesFromDeck(deck, eclipseState, *grid->c_grid(), param));
-    new_props.reset(new BlackoilPropsAdFromDeck(deck, eclipseState, *grid->c_grid()));
-
+    props.reset(new BlackoilPropertiesFromDeck(deck, eclipseState,
+                                               Opm::UgGridHelpers::numCells(grid),
+                                               Opm::UgGridHelpers::globalCell(grid),
+                                               Opm::UgGridHelpers::cartDims(grid),
+                                               Opm::UgGridHelpers::beginCellCentroids(grid),
+                                               Opm::UgGridHelpers::dimensions(grid), param));
+    new_props.reset(new BlackoilPropsAdFromDeck(deck, eclipseState, grid));
```

Comment:
As for the grid objects, we could also change the properties to use stack instead of heap, the original reason to use pointers does not exist anymore (actual class used depended on presence of a deck in older versions, but now we always must have a deck). Do you agree?